src/datastore/src/main/java/org/locationtech/geogig/geotools/data/GeogigFeatureStore.java

```diff
@@ -85,7 +85,7 @@ public void setTransaction(Transaction transaction) {
                 GeogigFeatureStore.this.setTransaction(transaction);
             }
         };
-        super.hints = delegate.getSupportedHints();
+        //super.hints = delegate.getSupportedHints();
```

Comment:
@DBlasby I asked in the GeoTools channel and it seems the best way forward here is to do a similar _double_ cast to set the Hints, similarly to how `getSupportedHints` does in ContentFeatureSource. Something like this:
`super.hints = (Set<Hints.Key>) (Set<?>)delegate.getSupportedHints();`
to essentially _double cast_ it back (i.e. reverse double cast)